model/canvas.cc

```diff
@@ -662,12 +662,58 @@ namespace minsky
   void Canvas::ungroupItem()
   {
     if (auto g=dynamic_cast<Group*>(item.get()))
-      {
+      {		  
         if (auto p=g->group.lock())
           {
-            p->moveContents(*g);
-            deleteItem();
-          }
+            if (!g->empty()) // minskly crashes if group empty and ungrouped subsequently. for ticket 1243
+              {  	    				 
+                // stash values of parameters in copied group, as they are reset for some unknown reason later on. for tickets 1243/1258
+                map<string,string> existingParms; 
+                for (auto& i: g->items) {
+                  auto v=i->variableCast(); 
+                  if (v && v->type()==VariableType::parameter) 
+                    existingParms.emplace(v->valueId(),v->init());
+                }
+		       
+                auto copyOfItems=g->items;
+                auto copyOfGroups=g->groups;				  
+               
+                p->moveContents(*g);
+                deleteItem();
+               
+                selection.clear();  // ensure no previous selection because ungrouped items will be left in selection afterwards.             
+               
+                // leave newly ungrouped items in selection
+                for (auto& i: copyOfItems) {
+                  selection.toggleItemMembership(i);
+                  // ensure that initial values of pasted parameters are correct. for ticket 1243/1258
+                  if (auto v=i->variableCast())
+                    if (v->type()==VariableType::parameter && !existingParms.empty()) 
+                      {
+                        auto it=existingParms.find(v->valueId());
+                        if (it!=existingParms.end()) v->init(it->second);
+                      }
+                }
+	           
+                selection.autoLayout();
+	           
+                if (!existingParms.empty()) existingParms.clear();
+	           
+                // Attach mouse focus only to first visible item in selection. For ticket 1098.      
+                for (auto& i: selection.items)
+                  if (i->visible())
+                    {
+                      setItemFocus(i);
+                      break;
+                    }
+                                   
+                if (!copyOfGroups.empty()) setItemFocus(copyOfGroups[0]);           
```

Comment:
This assumes that all groups will be visible. I think that is the case currently, but who knows.